lib/swoosh/x_test_assertions.ex

```diff
@@ -275,6 +295,96 @@ defmodule Swoosh.X.TestAssertions do
     do_flush_emails([])
   end
 
+  @doc ~S"""
+  Asserts multiple emails were sent.
+
+  You can pass a list of maps to match on specific params per email
+
+  ## Examples
+
+      iex> alias Swoosh.Email
+      iex> import Swoosh.TestAssertions
+
+      iex> emails = Enum.map(1..2, fn n -> Email.new(subject: "Hello, Avengers #{n}!") end)
+      iex> Swoosh.Adapters.Test.deliver_many(emails, [])
+
+      # assert a specific email was sent
+      iex> assert_emails_sent(emails)
+
+      # assert the list of emails with specific field(s) that were sent
+      iex> assert_email_sent([
+        %{subject: "Hello, Avengers 1!"},
+        %{subject: "Hello, Avengers 2!"},
+      ])
+  """
+  @spec assert_emails_sent() :: tuple | no_return
+  def assert_emails_sent do
+    assert_receive {:emails, _}
+  end
+
+  @spec assert_emails_sent([email_assertion()]) ::
+          :ok | tuple | no_return
+  def assert_emails_sent([%Swoosh.Email{} | _] = emails) do
+    assert_received {:emails, ^emails}
+  end
+
+  def assert_emails_sent([%{} | _] = params_map_list) do
+    assert_received {:emails, emails}
+
+    assert length(emails) == length(params_map_list)
+
+    emails
+    |> Enum.zip(params_map_list)
+    |> Enum.each(fn {email, params_map} ->
+      Enum.each(params_map, &assert_equal(email, &1))
+    end)
+  end
+
+  defp assert_equal(email, {:subject, %Regex{} = value}),
```

Comment:
Should we do the assertions below directly in the assert_emails_sent? I think it would be more explicity what we are testing.

Reply:
You can probably reuse these https://github.com/swoosh/swoosh/blob/f21726a81f7146a99b16fd43c929e6e0d3ee48bf/lib/swoosh/x_test_assertions.ex#L169-L209 and do similar things to https://github.com/swoosh/swoosh/blob/f21726a81f7146a99b16fd43c929e6e0d3ee48bf/lib/swoosh/x_test_assertions.ex#L123-L166. When I said "The existing implementation can serve as a good implementation guide" I meant the X version. Sorry I wasn't clear. 😢

Reply:
I think `has?` is missing a couple regex cases, comparing to `assert_equal`

Reply:
Sorry for the confusion :smile: 

I removed the assert_equal in order to use `has?` . Also , I implemented the regex case and a couple of tests for subject which was the only case I noticed comparing with assert_equal. Finally, I changed the implementation of the `assert_emails_sent`. Let me know what you think.